ISS-158592: Modularize the mobile SDK documentation

dhananjayj-dev opened the pull request:
This PR unifies the naming convention for Mobile SDKs across all platforms to enhance clarity and consistency. Additionally, the documentation is restructured into modular subpages for better navigation and maintainability. The changes are applied to the Fern documentation while the existing READMEs and codebase.

Issues Link:
/close [ISS-158592](https://app.devrev.ai/devrev/works/ISS-158592)
/close [ISS-160283](https://app.devrev.ai/devrev/works/ISS-160283)
/close [ISS-138840](https://app.devrev.ai/devrev/works/ISS-138840)

bojan commented on fern/docs/pages/developer-intro.mdx:
```diff
@@ -37,14 +37,34 @@ Numerous snap-ins are available on the DevRev [Marketplace](https://devrev.ai/ma
-    icon="fa-solid fa-message"
-    href="/sdks/mobile/android"
+    title="- DevRev SDK for Android"
+    icon="fa-brands fa-android"
+    href="/sdks/android/android"
+    />
+
+  <Card
+    title="- DevRev SDK for iOS"
+    icon="fa-brands fa-apple"
+    href="/sdks/ios/ios"
+    />
+
+  <Card
+    title="- DevRev SDK for React Native"
+    icon="fa-brands fa-react"
+    href="/sdks/react-native/react-native"
     />
+
     <Card
-    title="PLuG iOS SDK"
-    icon="fa-solid fa-message"
-    href="/sdks/mobile/ios"
+    title="- DevRev SDK for React Native expo"
+    icon="fa-brands fa-react"
+    href="/sdks/react-native-expo/react-native-expo"
+    />
+
+  <Card
+    title="- DevRev SDK for Cordova"
+    icon="fa-brands fa-js"
+    href="/sdks/cordova/cordova"
     />
 </Cards>
```
I believe something is off in the indentation of the cards.

bojan commented on fern/docs/pages/developer-intro.mdx:
```diff
@@ -37,14 +37,34 @@ Numerous snap-ins are available on the DevRev [Marketplace](https://devrev.ai/ma
-    title="PLuG iOS SDK"
-    icon="fa-solid fa-message"
-    href="/sdks/mobile/ios"
+    title="- DevRev SDK for React Native expo"
```
Two things here:
- @rohan-devrev do we need to separate the guides for RN and Expo?
- The framework is named `Expo`, not `React Native expo`.

bojan commented on fern/docs/pages/plug-sdk/android/quickstart.mdx:
```diff
@@ -0,0 +1,144 @@
+# Quickstart Guide
```
As we mentioned in a few other PRs the titles should have a `Sentence case`.
Let's check it everywhere.

bojan commented on fern/docs/pages/plug-sdk/android/quickstart.mdx:
Will we add the samples in a separate PR?

dhananjayj-dev replied:
Added the sample app section.

bojan commented on fern/docs/pages/plug-sdk/cordova/quickstart.mdx:
```diff
@@ -0,0 +1,32 @@
+# Quickstart Guide
```
Sentence case.

bojan commented on fern/docs/pages/plug-sdk/index.mdx:
Check the comment about the indentation and Expo.

bojan commented on fern/docs/pages/plug-sdk/ios/quickstart.mdx:
```diff
@@ -0,0 +1,73 @@
+# Quickstart Guide
```
Sentence case.

bojan commented on fern/docs/pages/plug-sdk/react-native-expo/quickstart.mdx:
```diff
@@ -0,0 +1,32 @@
+# Quickstart Guide
```
Sentence case.

bojan commented on fern/docs/pages/plug-sdk/react-native/quickstart.mdx:
```diff
@@ -0,0 +1,25 @@
+# Quickstart Guide
```
Sentence case.

bojan reviewed (commented):
This look amazing and very promising.

The PR title is too big, let's try to shorten it, e.g. Modularize the mobile SDK documentation.

Also, do we actually need to have the files under the `plug-sdk` folder? Can we use `mobile-sdk`?

@rohan-devrev let's decide whether we need to have separate React Native and Expo guides. I'd name the guide DevRev SDK for React Native and Expo.

bojan commented:
The React Native guide doesn't seem to be up-to-date as it seems to have conflicts.

dhananjayj-dev commented:
> The React Native guide doesn't seem to be up-to-date as it seems to have conflicts.

I'll update this with the latest change.

dhananjayj-dev commented:
> This look amazing and very promising.
> 
> The PR title is too big, let's try to shorten it, e.g. Modularize the mobile SDK documentation.
> 
> Also, do we actually need to have the files under the `plug-sdk` folder? Can we use `mobile-sdk`?
> 
> @rohan-devrev let's decide whether we need to have separate React Native and Expo guides. I'd name the guide DevRev SDK for React Native and Expo.

I have moved the files under `sdks -> mobile` namespace. Removed the Expo documentations as it's same as React Native.

bojan reviewed (approved):
This is amazing work! Let's wait for @bc-devrev's review, so we can merge it and release Cordova! 💪🏼

bc-devrev commented on fern/docs/pages/sdks/mobile/react-native/quickstart-expo.mdx:
````diff
@@ -0,0 +1,50 @@
+	npx expo prebuild --clean
+	```
+
+## Setting up the DevRev SDK
````
```suggestion
## Set up the DevRev SDK
```

bc-devrev commented on fern/docs/pages/sdks/mobile/react-native/quickstart-expo.mdx:
```diff
@@ -0,0 +1,50 @@
+
+## Setting up the DevRev SDK
+
+### Step 1: Credentials
+
+1. Open the DevRev web app at [https://app.devrev.ai](https://app.devrev.ai).
+1. Go to the **Settings** page.
+1. Then open the **PLuG Settings** page, and copy the value under **Your Unique App ID**.
+
+### Step 2: Configuration
+
+> [!IMPORTANT]
+> The SDK must be configured before you can use any of its features.
+
+Once you have the credentials, you can configure the DevRev SDK in your app. The SDK will be ready to use once you have called the configuration method:
```
This is a short procedure. There's no need to split it into two pieces.
```suggestion

1. In the DevRev app, go to **Settings** > **PLuG Settings**.
1. Copy the value under **Your Unique App ID**.
1. Configure the DevRev SDK in your app by calling the configuration method:
```

bc-devrev commented on fern/docs/pages/sdks/mobile/react-native/quickstart-expo.mdx:
````diff
@@ -0,0 +1,50 @@
+
+Once you have the credentials, you can configure the DevRev SDK in your app. The SDK will be ready to use once you have called the configuration method:
+
+```typescript
````
Indent 3 spaces.

bc-devrev reviewed (approved):
The name for the web SDK was inconsistent so I fixed it to be "PLuG SDK for Web" everywhere.

I'm going to go ahead and approve but there are a few superficial problems that need to be fixed either in this PR or a separate one.
- Headings must not be gerunds ("Set up the SDK", not "Setting up the SDK"). https://www.notion.so/devrev/How-should-instructions-be-written-dac658b248d94db6a0c301079cffbbf5?pvs=4
- Headings must be in sentence case. https://www.notion.so/devrev/Which-words-in-a-heading-title-or-label-should-be-capitalized-81984d54dd414af5b343bce1327c05bd?pvs=4
- The "Migration guide" content doesn't fit comfortably in the tables. 
  ![image](https://github.com/user-attachments/assets/1cf4ae0a-8873-490c-a730-f987a4f66cd2)
  Something like this might work better:
  ```
  **Session properties**
  - `UserExperior.setUserProperties(userProperties)`
  - `DevRev.addSessionProperties(properties, successCallback, errorCallback)`
  - `DevRev.clearSessionProperties()`
  ```

bc-devrev commented:
> Also, do we actually need to have the files under the `plug-sdk` folder? Can we use `mobile-sdk`?

You can create new folders as necessary. If you change any file paths, you'll need to update them in the `versions/public.yml` file. If you change any URLs, be sure to add redirects in the `docs.yml` file.

dhananjayj-dev commented:
> The name for the web SDK was inconsistent so I fixed it to be "PLuG SDK for Web" everywhere.
> 
> I'm going to go ahead and approve but there are a few superficial problems that need to be fixed either in this PR or a separate one.
> 
> * Headings must not be gerunds ("Set up the SDK", not "Setting up the SDK"). https://www.notion.so/devrev/How-should-instructions-be-written-dac658b248d94db6a0c301079cffbbf5?pvs=4
> * Headings must be in sentence case. https://www.notion.so/devrev/Which-words-in-a-heading-title-or-label-should-be-capitalized-81984d54dd414af5b343bce1327c05bd?pvs=4
> * The "Migration guide" content doesn't fit comfortably in the tables.
>   ![image](https://private-user-images.githubusercontent.com/117322957/443411371-1cf4ae0a-8873-490c-a730-f987a4f66cd2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDczMDY5NDAsIm5iZiI6MTc0NzMwNjY0MCwicGF0aCI6Ii8xMTczMjI5NTcvNDQzNDExMzcxLTFjZjRhZTBhLTg4NzMtNDkwYy1hNzMwLWY5ODdhNGY2NmNkMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxNVQxMDU3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzRmODNmMTVkMThjMjg3NjQwYWNmMzVhYTQ0NWI5ZTI4Mjc3ZjA1NTk5MDQ3OGZlMzljMDA4OTViNzMzNzdiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Nxfw1go9i-lzEy-Gfq7ZceR_FxVXO48qebn3zrhQQhc)
>   Something like this might work better:
>   ```
>   **Session properties**
>   - `UserExperior.setUserProperties(userProperties)`
>   - `DevRev.addSessionProperties(properties, successCallback, errorCallback)`
>   - `DevRev.clearSessionProperties()`
>   ```

I have addressed all the other comments except for the migration guide. Can we keep it as is since it fits best in a table format and can also be viewed in full screen?

cc @bojan

bc-devrev commented:
> I have addressed all the other comments except for the migration guide. Can we keep it as is since it fits best in a table format and can also be viewed in full screen?

I'll leave it up to you what to do about the migration guides. The PR is approved so you can merge when you're ready.
